opendatahub-io/argo-workflows
# [Snyk] Fixing go.mod snyk CVEs

## Description of your changes:


Fixing go.mod dependencies/packages related snyk CVEs.
Imported my fork in synk and made sure the relevant CVEs were fixed with this update.
Added a replace block for each CVE with the fix version suggested by Snyk.
Ran `go mod tidy`. Wherever go version compatibility issues came up, ran `go mod tidy -go=1.16 && go mod tidy -go=1.17`
Pushed changes to my fork and ensured the CVEs were resolved.
Also executed a `go build .` locally and ensure it doesn't return any errors.

These updates resolve 24/27 Snyk High Level errors for the go.mod.

## Testing instructions

Pull the changes in this PR, push them to your fork's main (make sure the main branch of the fork is in sync with odh-io/argo-workflows main). Import your fork in snyk and verify.
